packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -0,0 +1,373 @@
+      ...orbitNetworks
+    ]
+  }, [debouncedNetworkSearched, networks])
+
+  const rowRenderer = useCallback(
+    (virtualizedProps: ListRowProps) => {
+      const networkOrChainTypeInfo =
+        networksToShowWithChainTypeInfo[virtualizedProps.index]
+      if (!networkOrChainTypeInfo) {
+        return null
+      }
+
+      // Chain Type Info row
+      if (typeof networkOrChainTypeInfo === 'object') {
+        const isCoreGroup = networkOrChainTypeInfo.name === 'CORE CHAINS'
+        return (
+          <div
+            key={networkOrChainTypeInfo.name}
+            style={virtualizedProps.style}
+            className={twMerge(
+              'px-6 py-3',
+              !isCoreGroup &&
+                'before:-mt-3 before:mb-3 before:block before:h-[1px] before:w-full before:bg-black/30 before:content-[""]'
+            )}
+          >
+            <p className="text-sm text-dark">{networkOrChainTypeInfo.name}</p>
+            {networkOrChainTypeInfo.description && (
+              <p className="mt-2 text-xs">
+                {networkOrChainTypeInfo.description}
+              </p>
+            )}
+          </div>
+        )
+      }
+
+      return (
+        <NetworkRow
+          key={networkOrChainTypeInfo}
+          style={virtualizedProps.style}
+          chainId={networkOrChainTypeInfo}
+          onClick={onNetworkRowClick}
+          close={close}
+        />
+      )
+    },
+    [close, networksToShowWithChainTypeInfo, onNetworkRowClick]
+  )
+
+  const onSearchInputChange = function (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) {
+    setErrorMessage('')
+    setNetworkSearched(event.target.value)
+  }
+
+  return (
+    <div className="flex flex-col gap-4">
+      <SearchPanelTable
+        searchInputPlaceholder="Search a network name"
+        searchInputValue={networkSearched}
+        onSearchInputChange={onSearchInputChange}
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        errorMessage={errorMessage}
+        rowCount={networksToShowWithChainTypeInfo.length}
+        rowHeight={getRowHeight}
+        rowRenderer={rowRenderer}
+        listRef={listRef}
+      />
+      <div className="flex justify-between pb-2">
+        <TestnetToggle
+          className={testnetToggleClassNames}
+          label="Testnet mode"
+        />
+        <AddCustomOrbitChainButton />
+      </div>
+    </div>
+  )
+}
+
+export const NetworkSelectionContainer = ({
+  children,
+  buttonClassName,
+  buttonStyle,
+  onChange
+}: {
+  children: React.ReactNode
+  buttonClassName: string
+  buttonStyle?: CSSProperties
+  onChange: (value: Chain) => void
+}) => {
+  const { isTestnetMode } = useIsTestnetMode()
+
+  const supportedNetworks = getSupportedChainIds({
+    includeTestnets: isTestnetMode
+  })
+
+  const { isSmartContractWallet, isLoading: isLoadingAccountType } =
+    useAccountType()
+
+  const coreNetworks = useMemo(
+    () =>
+      supportedNetworks.filter(
```

Comment:
nit: because we use it often now maybe we can add `isCoreChain` to `isNetwork`. This way we unify the definition of the core chain, because before we used `!isOrbitChain` to define it